app/controllers/application_controller.rb

```diff
@@ -22,6 +22,39 @@ class ApplicationController < ActionController::API
 
   private
 
+  set_current_tenant_through_filter
+  before_action :set_the_current_tenant
+  after_action  :unset_the_current_tenant
+
+  def set_the_current_tenant
+    return unless Tenant.tenancy_enabled?
+
+    tenant = Tenant.find_by(:external_tenant => user_identity)
```

Comment:
Can we just let this blow up naturally with an ActiveRecord::RecordNotFound and go through the normal path instead of building our own error document?